lib/puppet/resource/catalog.rb

```diff
@@ -366,18 +369,14 @@ def resource(type, title = nil)
       # an instance has to be created in order to construct the unique key used when
       # searching for aliases, or when app_management is active and nothing is found in
       # which case it is needed by the CapabilityFinder.
-# PUP-4947
       res = nil
-#      app_mgnt = Puppet[:app_management]
-#      if app_mgnt || !@aliases.empty?
       res = Puppet::Resource.new(type, title, { :environment => @environment_instance })
 
-        # No need to build the uniqueness key unless there are aliases
-      result = @resource_table[[type_name, res.uniqueness_key].flatten] unless @aliases.empty?
-# PUP-4947
-#      end
+      # Must check with uniqueness key because of aliases or if resource transforms title in title
+      # to attribute mappings.
+      result = @resource_table[[type_name, res.uniqueness_key].flatten]
```

Comment:
No impact on performance done. Note that that only doing the uniqueness test if there are aliases is wrong since the uniqueness key is needed for all resources that have one that is different from the title. Not even if the uniqueness key contains just a single entry can you be sure that it is not different.